src/nomad_simulations/model_system.py

```diff
@@ -967,6 +977,25 @@ def normalize(self, archive, logger) -> None:
         if is_not_representative(self, logger):
             return
 
+        # Traversing the System Hierarchy
+        # TODO consider moving to utils and passing generic set of functions to execute at each branch level
+        def set_branch_composition(system, subsystems, atom_labels):
+            if subsystems is None:
+                subsystem_labels = [np.array(atom_labels)[system.get("atom_indices")]]
+            else:
+                subsystem_labels = [subsystem.get("branch_label") for subsystem in subsystems]
```

Comment:
```python
subsystem_labels = [subsystem.branch_label for subsystem in subsystems if subsystem.branch_label is not None]
```
error handling for when `branch_label` is not set.

Reply:
Actually, what do you think about something like this:
```python
subsystem_labels = [subsystem.get("branch_label", "Unknown") for subsystem in subsystems]
```

This way you can see if branches exist even if they don't have labels

Reply:
Yeah, that's another option. Then I agree, we can use `.get()` in this case to indicate a default value `"Unknown"`.